pages/search.tsx

```diff
@@ -70,25 +92,6 @@ export default function Search({ serverSearchByText }: ServerProps) {
     })()
   }, [filterSearchByText, filterSearchByType, filterPage])
 
-  /* Client-Side Queries */
-
-  const clientQueryPodcasts = async () => {
-    let response = {
-      data: [[], 0]
-    } as any
-
-    if (filterSearchByText) {
-      const finalQuery = {
-        ...(filterPage ? { page: filterPage } : {}),
-        ...(filterSearchByType ? { searchBy: filterSearchByType } : {}),
-        searchText: filterSearchByText
-      }
-      response = await getPodcastsByQuery(finalQuery)
-    }
-
-    return response
-  }
-
```

Comment:
Can you explain why you moved the function outside of the render function? We've been putting the helper functions within the render function across all the files. Not sure there is any good reason for it, it's just how we've been doing it. CC @kreonjr

Reply:
For three reasons

1) It is easier to actually share with other files and encourages a better interface due to requiring all values be passed in rather than everything being available via closure
2) Technically having it in the render function is additional overhead because every time the component renders it needs to create a new instance of the function. This can be avoid by using `useCallback`
3) If you ever want to test either the network calls or components, having them separate is easier

Reply:
@RyanHirsch alright thanks for the answer. I started the project by writing the functions outside of the render, then started moving them all into the render...now I have to move them out again 😅 not going to rewrite everything now, but going forward I'll keep it in mind.